openmls/src/treesync/treekem.rs

```diff
@@ -44,13 +44,11 @@ impl<'a> TreeSyncDiff<'a> {
         group_context: &[u8],
         exclusion_list: &HashSet<&LeafIndex>,
         key_package: KeyPackage,
-    ) -> Result<UpdatePath, TreeKemError> {
+    ) -> Result<UpdatePath, LibraryError> {
         let copath_resolutions = self.copath_resolutions(self.own_leaf_index(), exclusion_list)?;
 
         // There should be as many copath resolutions.
-        if copath_resolutions.len() != path.len() {
-            return Err(TreeKemError::PathLengthError);
-        }
+        debug_assert_eq!(copath_resolutions.len(), path.len());
```

Comment:
Sorry to be pedantic about this, but can you add a doc comment saying that this function assumes that the input `path` has the same length as the copath resolution of the own leaf?